src/component/tooltip/TooltipView.ts

```diff
@@ -752,12 +805,50 @@ class TooltipView extends ComponentView {
             html = formatter(params, asyncTicket, callback);
         }
 
-        tooltipContent.setContent(html, markers, tooltipModel);
-        tooltipContent.show(tooltipModel);
-
         this._updatePosition(
             tooltipModel, positionExpr, x, y, tooltipContent, params, el
         );
+        tooltipContent.setContent(html, markers, tooltipModel, nearPoint.color, positionExpr);
+        tooltipContent.show(tooltipModel, nearPoint.color);
+
+    }
+
+    _getNearestPoint(
+        point: number[],
+        tooltipDataParams: TooltipDataParams | TooltipDataParams[],
+        coord?: CoordinateSystem
+    ): {
+        x: number;
+        y: number;
+        color: ZRColor;
+    } {
+        let dim = '';
+        if (coord && coord.type === 'cartesian2d') {
+            dim = coord.getBaseAxis().dim;
+        }
+        if (!zrUtil.isArray(tooltipDataParams)) {
+            if (!tooltipDataParams.position) {
+                return {
+                    x: point[0],
+                    y: point[1],
+                    color: tooltipDataParams.color || tooltipDataParams.borderColor
+                };
+            }
+            return {
+                x: point[0],
+                y: tooltipDataParams.position[1],
+                color: tooltipDataParams.color || tooltipDataParams.borderColor
+            };
+        }
+
+        const posIndex = +(dim === 'x');
```

Comment:
Not a good way to determine axis dimension. only suitable for cartesian.
We could use Euclidean distance to avoid that axis dimension determine.